datafusion/src/execution/context.rs

```diff
@@ -573,18 +578,10 @@ impl SessionContext {
 
         let state = self.state.lock();
         let catalog = if state.config.information_schema {
-            let is = state
-                .catalog_list
-                .catalog("datafusion")
-                .unwrap()
-                .schema("information_schema");
-            match is {
-                Some(_) => catalog,
-                None => Arc::new(CatalogWithInformationSchema::new(
-                    Arc::downgrade(&state.catalog_list),
-                    catalog,
-                )),
-            }
+            Arc::new(CatalogWithInformationSchema::new(
```

Comment:
This is the fix -- I did not understand what this special case was for (and all the tests seem to pass without it). It seems like this code will basically ignore the `catalog` parameter passed in if it finds an existing `information_schema` which doesn't seem correct to me

@matthewmturner  do you remember why it was added in https://github.com/apache/arrow-datafusion/pull/1959?